File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/injection/module/ActivityModule.java
package com.national.security.community.injection.module;

import com.national.security.community.injection.scope.ActivityScope;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;

import dagger.Module;
import dagger.Provides;


@Module
public class ActivityModule {
    private RxAppCompatActivity mActivity;

    public ActivityModule(RxAppCompatActivity activity) {
        this.mActivity = activity;
    }

    @Provides
    @ActivityScope
    RxAppCompatActivity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/com/jmd_news_kotlin/di/module/ActivityModule.kt
package com.jmd_news_kotlin.di.module

import android.app.Activity
import com.jmd_news_kotlin.di.scope.ActivityScope
import dagger.Module
import dagger.Provides
import javax.inject.Inject

@Module
class ActivityModule @Inject constructor(activity: Activity) {

    private var mActivity: Activity = activity

    @Provides
    @ActivityScope
    fun providerActivity(): Activity {
        return mActivity
    }
}